noxfile.py

```diff
@@ -78,6 +81,12 @@ def run_integration(session):
     """Run the integration tests."""
     set_environment_variables(PYBAMM_ENV, session=session)
     session.install("setuptools", silent=False)
+    if (
+        "CI" in os.environ
+        and sys.version_info[:2] == (3, 12)
+        and sys.platform == "linux"
```

Comment:
The annotations will be same for all platforms, right?

Reply:
Yes, they will be the same, and we found that not restricting to one platform and one Python version introduces redundant annotations (14 annotations for one test failure). This won't help where tests fail on, say, Python 3.11 but pass on Python 3.12, but it's the best one we have for now – I think

Reply:
Sounds Good.